src/pages/Token/Modals/TokenPageModal.tsx

```diff
@@ -54,6 +55,8 @@ const TokenPageModal = ({ onClose, onFinish, marketType, offer, token }: TTokenP
         return PurchaseModal;
       case MarketType.transfer:
         return TransferModal;
+      case MarketType.share:
```

Comment:
put this modal separately, those modals are for token actions (transfer/sell/etc), while share modal is on it's own